analyze.js

```diff
@@ -0,0 +1,242 @@
+moveFile(latencyHistogramFile, path.join(assetsDir, latencyHistogramFile));
+
+const serverRPS = {};
+servers.forEach((server) => {
+  serverRPS[server] = avgReqSecs[server];
+});
+
+const sortedServers = Object.keys(serverRPS).sort(
+  (a, b) => serverRPS[b] - serverRPS[a]
+);
+const lastServer = sortedServers[sortedServers.length - 1];
+const lastServerReqSecs = avgReqSecs[lastServer];
+
+const resultsFile = "results.md";
+
+try {
+  if (!fs.existsSync(resultsFile) || fs.readFileSync(resultsFile, 'utf8').trim() === '') {
+    fs.writeFileSync(resultsFile, `<!-- PERFORMANCE_RESULTS_START -->
+
+| Query | Server | Requests/sec | Latency (ms) | Relative |
+|-------:|--------:|--------------:|--------------:|---------:|`);
+  }
+} catch (error) {
+  console.error(`Error initializing results file: ${error.message}`);
+}
+
+let resultsTable = "";
+
+if (whichBench === 1) {
+  resultsTable += `\n| ${whichBench} | \`{ posts { id userId title user { id name email }}}\` |`;
+} else if (whichBench === 2) {
+  resultsTable += `\n| ${whichBench} | \`{ posts { title }}\` |`;
+} else if (whichBench === 3) {
+  resultsTable += `\n| ${whichBench} | \`{ greet }\` |`;
+}
+
+sortedServers.forEach((server) => {
+  const formattedReqSecs = avgReqSecs[server].toLocaleString(undefined, {
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  });
+  const formattedLatencies = avgLatencies[server].toLocaleString(undefined, {
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  });
+  const relativePerformance = (avgReqSecs[server] / lastServerReqSecs).toFixed(2);
+
+  resultsTable += `\n|| [${formattedServerNames[server]}] | \`${formattedReqSecs}\` | \`${formattedLatencies}\` | \`${relativePerformance}x\` |`;
+});
+
+try {
+  fs.appendFileSync(resultsFile, resultsTable);
+} catch (error) {
+  console.error(`Error appending to results file: ${error.message}`);
+}
+
+if (whichBench === 3) {
+  try {
+    fs.appendFileSync(resultsFile, "\n\n<!-- PERFORMANCE_RESULTS_END -->");
+
+    const finalResults = fs
+      .readFileSync(resultsFile, "utf-8")
+      .replace(/\\/g, '');  // Remove backslashes
+
+    const readmePath = "README.md";
+    let readmeContent = fs.readFileSync(readmePath, "utf-8");
+    const performanceResultsRegex =
+      /<!-- PERFORMANCE_RESULTS_START -->[\s\S]*<!-- PERFORMANCE_RESULTS_END -->/;
+    if (performanceResultsRegex.test(readmeContent)) {
+      readmeContent = readmeContent.replace(
+        performanceResultsRegex,
+        finalResults
+      );
+    } else {
+      readmeContent += `\n${finalResults}`;
+    }
+    fs.writeFileSync(readmePath, readmeContent);
+    console.log("README.md updated successfully");
+  } catch (error) {
+    console.error(`Error updating README: ${error.message}`);
+  }
+}
+
+resultFiles.forEach((file) => {
+  try {
+    fs.unlinkSync(file);
+  } catch (error) {
+    console.error(`Error deleting file ${file}: ${error.message}`);
+  }
+});
```

Comment:
- [ ] Split this file into smaller ones. Create a `parser.rs` that parses the generated data. It takes in input as string and produces a parsed output. 
- [ ] Make this code type-safe using Typescript. Don't use `any` anywhere.
- [ ] Drop all the server-specific code from the script.

Reply:
okay

Reply:
parser.rs will be on a new pr...